### ISSUE TITLE
[entropy_src] Delay ht_done_pulse_i for the main SM by 1 cycle 

### ISSUE THREAD

**Issue by h-filali** [lowRISC/opentitan#23773]
### Description

Moving the 1-4-bit packer back in the pipeline #21626 introduced an issue where the `ht_done_pulse_i` can go high 1 cycle too early. This in combination with some other recent changes caused some unwanted words to be absorbed by SHA3.

When `ht_done_pulse_i` goes high this causes `process_i` for SHA3 to go high, which in turn leads to SHA3 not accepting any new inputs. If these signals go high 1 cycle too early SHA3 doesn't absorb the last word. The result were some test sequences that were failing due to a mismatch in the produced seeds.

I only observed it happening for the single lane mode, where one of the 4 random bit lanes is selected. This might be because the `post_ht` FIFO still receives the entropy without delay in normal operation.

Unfortunately I can't post a failing seed here for the `entropy_src_rng` test, since I had to change a lot in the DV environment to discover this bug.

**Comment by vogelpi** [comment]
FYI @moidx , @johannheyszl 

@h-filali and I had a closer look at this. The current behavior of the RTL (without the proposed fix in #23775) is:
- The hardware modes using all four lanes work as expected.
- The firmware override modes using all four lanes or 1 lane only work as expected.
- The hardware mode using a single lane only has the following issue:
  - The health test done and failure pulses are signaled one clock cycle too early. This means the last 64-bit word accumulated in the precon packer FIFO (see [block diagram](https://opentitan.org/book/hw/ip/entropy_src/doc/theory_of_operation.html#block-diagram)) isn't absorbed by the SHA3 conditioner.
  - For the very first seed, the conditioner adds more padding bits. So for the very first seed, we use less bits of entropy than expected. 
  - When starting the next seed, the conditioner loads the last 64-bit word of the previous health test window (still waiting in the precon FIFO) into the re-initialized conditioner. The last 64-bit of the new health test window is then again missed, but the number of bits used to produce the seed is the as expected.
  - This means the health test window is misaligned compared to the seed.

To summarize, ideally we could absorb @h-filali 's fix but it's probably not a complete deal breaker we we can't because:
- Only the hardware mode with a single lane is affected but not the hardware mode with 4 lanes and not the fallback firmware override modes.
- The defined amount of entropy is used for every seed except for the first one (64 bit less than defined) but the health test window and the seed window is misaligned by 64 bits.

**Comment by vogelpi** [comment]
We've discussed this issue in the tape-out sync yesterday (2024-06-24) and reached consensus to:

1. Not merge this / integrate this as ECO now. The reasons are mostly that i) the issue doesn't affect the firmware override modes (our fallback mode) ii) the window size can be increased to compensate for the first seed using slightly less entropy iii) CSRNG / EDN can be reconfigured / reseeded to drop the first seed using less entropy.
2. Integrate the change post M5 after the TO branch has been taken to allow this bug fix for future releases / revisions of the design.

Integrating this improvement after taking the TO branch should be easy as it doesn't involve any software changes which could break the compatibility with Earlgrey-PROD A1.
